### PR TITLE
Make Entitlements in Templates for MacCatalyst more consistent between debug and release configs.

### PR DESCRIPTION
**Pull request by dustin-wojciechowski**
### Description of Change
Make Entitlements in MacCatalyst Templates more consistent between debug and release configs. 

See [teams conversation](https://teams.microsoft.com/l/message/19:989ffa44998147aca4ceaf7482967668@thread.skype/1696009594958?tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47&groupId=0056f60b-301f-43ac-bbcf-f356d3c42c92&parentMessageId=1696009594958&teamName=VS%20Client%20Experiences&channelName=MAUI%20%F0%9F%8C%BA&createdTime=1696009594958) for more context (MSFT only)

### CODE REVIEW

**Review by Eilon — COMMENTED** [review]
This definitely aligns with what I was looking for in our discussion. Can you confirm that you tested locally on a Mac in debug/release and native/hybrid to make sure there's nothing unexpected about using Sandbox in debug?

**Comment by dustin-wojciechowski** [comment]
> This definitely aligns with what I was looking for in our discussion. Can you confirm that you tested locally on a Mac in debug/release and native/hybrid to make sure there's nothing unexpected about using Sandbox in debug?

Yes, I tested locally with two new maui-mobile and maui-blazor projects and there were no issues running the samples.

**Review by jsuarezruiz — DISMISSED** [review]

**Review by mattleibow — COMMENTED** [review]
Is it possible to remove all the conditions in the templates and make all the items in the plist? For the dev tools, we can just comment it out and a user can uncomment if they want it. I would like to reduce the things in the csproj file.

**Review by PureWeen — CHANGES_REQUESTED** [review]
Currently this PR is targeting main (SR1) with a GA milestone. 
Can you please update the Milestone to SR1 or update the targeted branch to net8.0.

**Comment by dustin-wojciechowski** [comment]
> Currently this PR is targeting main (SR1) with a GA milestone. Can you please update the Milestone to SR1 or update the targeted branch to net8.0.

Changed targeted branch to net8.0.

File: src/Templates/src/templates/maui-blazor/MauiApp.1.csproj [hunk 1]
```diff
@@ -65,12 +65,7 @@
...
-  <PropertyGroup Condition="$(TargetFramework.Contains('-maccatalyst')) and '$(Configuration)' == 'Debug'">
-    <CodesignEntitlements>Platforms/MacCatalyst/Entitlements.Debug.plist</CodesignEntitlements>
-  </PropertyGroup>
-
-  <PropertyGroup Condition="$(TargetFramework.Contains('-maccatalyst')) and '$(Configuration)' == 'Release'">
-    <CodesignEntitlements>Platforms/MacCatalyst/Entitlements.Release.plist</CodesignEntitlements>
-    <UseHardenedRuntime>true</UseHardenedRuntime>
+  <PropertyGroup Condition="$(TargetFramework.Contains('-maccatalyst'))">
+    <CodesignEntitlements>Platforms/MacCatalyst/Entitlements.plist</CodesignEntitlements>
   </PropertyGroup>
```
Comment (mattleibow) [inline src/Templates/src/templates/maui-blazor/MauiApp.1.csproj:70]: This should not be necessary as the globs will add it:

https://github.com/dotnet/maui/blob/81fd8f6a9bd5ed5d4b41bff5f370801323c90050/src/Controls/src/Build.Tasks/nuget/buildTransitive/netstandard2.0/Microsoft.Maui.Controls.SingleProject.targets#L53

File: src/Templates/src/templates/maui-mobile/MauiApp.1.csproj [hunk 2]
```diff
@@ -63,12 +63,7 @@
...
-  <PropertyGroup Condition="$(TargetFramework.Contains('-maccatalyst')) and '$(Configuration)' == 'Debug'">
-    <CodesignEntitlements>Platforms/MacCatalyst/Entitlements.Debug.plist</CodesignEntitlements>
-  </PropertyGroup>
-
-  <PropertyGroup Condition="$(TargetFramework.Contains('-maccatalyst')) and '$(Configuration)' == 'Release'">
-    <CodesignEntitlements>Platforms/MacCatalyst/Entitlements.Release.plist</CodesignEntitlements>
-    <UseHardenedRuntime>true</UseHardenedRuntime>
+  <PropertyGroup Condition="$(TargetFramework.Contains('-maccatalyst'))">
+    <CodesignEntitlements>Platforms/MacCatalyst/Entitlements.plist</CodesignEntitlements>
   </PropertyGroup>
```
Comment (mattleibow) [inline src/Templates/src/templates/maui-mobile/MauiApp.1.csproj:68]: Same as above.

**Review by mattleibow — CHANGES_REQUESTED** [review]
I think this is almost ready, just need to remove the item in the csproj as it is already added.

**Review by mattleibow — APPROVED** [review]

**Review by PureWeen — APPROVED** [review]

**Review by Eilon — APPROVED** [review]
Much cleaner!!

**Pull request merged**